lib/stateful/mongoid.rb

```diff
@@ -50,6 +50,13 @@ def define_state_attribute(options)
                 scope scope_name, -> { where(name.in => states) }
               end
             end
+
+            # add tracked fields for this state
+            if info.tracked
+              field("#{info.name}_at", type: Time)
+              field("#{info.name}_by", type: User) if defined?(User)
```

Comment:
Should probably also only declare this field is `User.respond_to?(:current)` otherwise its kind of a dead field.